server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java

```diff
@@ -154,6 +154,10 @@ private boolean assertConsumesSupportedParams(@Nullable Set<String> supported, R
             supportedAndCommon.removeAll(RestRequest.INTERNAL_MARKER_REQUEST_PARAMETERS);
             final var consumed = new TreeSet<>(request.consumedParams());
             consumed.removeAll(RestRequest.INTERNAL_MARKER_REQUEST_PARAMETERS);
+            // Add exclusions for response parameters since they may not always be consumed for every request.
+            final var unconsumedResponseParams = new HashSet<>(responseParams(request.getRestApiVersion()));
+            unconsumedResponseParams.removeAll(consumed);
+            supportedAndCommon.removeAll(unconsumedResponseParams);
```

Comment:
I think this'll permit a case where we declare a response parameter but don't include it in the supported parameters, which would be a bug. Could we instead count the response parameters in `consumed` (they're implicitly consumed since they're made available to the response rendering)?